torch/_functorch/compilers.py

```diff
@@ -116,22 +118,51 @@ def nop(fx_g: fx.GraphModule, _) -> Callable:
     return fx_g
 
 class DebugInterpreter(fx.Interpreter):
+    def run(self, *args):
+        self.symbol_mapping = bind_symbols(self.module, *args)
+        super().run(*args)
+
     def run_node(self, n):
-        # TODO: This will fail once we start caching in AOTAutograd
-        # again, because we need to remap SymInts to their new values
-        # in the presence of dynamism
+        import sympy
```

Comment:
can we pull this up to toplevel?

Reply:
No, because we still have problems with sympy being a hard dep internally. Actually, your recent ShapeEnv PR might get reverted because it starts unconditionally importing sympy again; dev infra is trying to figure out if they can forward fix it right now.

Reply:
I think that PR guards it with an import check